assets/javascripts/test_result.js

```diff
@@ -600,17 +600,21 @@ function setCurrentPreviewFromStepLinkIfPossible(stepLink) {
 }
 
 function githashToLink(value, repo) {
-  var hashes_regex = /(\b[0-9a-f]{9}\b)/gms;
-  var githashes = value.match(hashes_regex);
-  var oneLineStatRegex = /\<a[^]+/gms;
-  var statRegex = /(\<a.+?)(?=(\<a))/gms;
-  if (!Array.isArray(githashes)) return;
-  for (let i = 0; i < githashes.length; i++) {
-    value = value.replace(githashes[i], githashes[i].link(repo + githashes[i]));
-    if (githashes.length == 1) return value.match(oneLineStatRegex);
+  const log_items = value.split(/(?=^[0-9a-f])/gm);
+  log_items.shift();
+  const commits = [];
+  for (let i = 0; i < log_items.length; i++) {
+    const commit = {};
+    const item = log_items[i];
+    const match = item.match(/^([0-9a-f]{9}) (.*)/);
+    const sha = match[1];
+    commit['sha'] = sha;
+    commit['link'] = sha.link(repo + sha);
+    commit['msg'] = match[2];
+    commit['stat'] = item.match(/^ .*/gm);
+    commits.push(commit);
```

Comment:
```suggestion
    commits.push({sha: sha, ...});
```
Instead of the extra variable and assigning properties one-by-one.